### PR TITLE
Feat/combined outcomes

### PR DESCRIPTION
**Pull request by kirilklein**
## Summary by CodeRabbit

- **New Features**
  - Introduced support for defining and detecting combination outcomes based on temporal relationships between multiple events (e.g., identifying when one event occurs within a specific time window before or after another event).
  - Added configuration options to specify the time window and timestamp source for combined outcomes.
  - Added a new combined outcome definition `TEST_COMBINED` for testing purposes.

- **Bug Fixes**
  - Improved handling of empty or missing data during outcome matching to ensure robust processing.

- **Tests**
  - Expanded and enhanced test coverage for outcome matching, including complex combination scenarios and edge cases.

### CODE REVIEW

File: corebehrt/modules/cohort_handling/outcomes.py [hunk 1]
```diff
@@ -32,27 +43,57 @@ def __call__(
...
-            timestamps[ABSPOS_COL] = timestamps[ABSPOS_COL].astype(int)
-            timestamps[PID_COL] = timestamps[PID_COL].astype(int)
+                types = attrs["type"]
+                matches = attrs["match"]
+                if types == "patients_info":
+                    timestamps = self.match_patient_info(patients_info, matches)
+                else:
+                    timestamps = self.match_concepts(
+                        concepts_plus, types, matches, attrs
+                    )
```
[the remark this answers is not in the document]
Comment (kirilklein) [reply corebehrt/modules/cohort_handling/outcomes.py:67]: Maybe we should remove the patients_info option alltogether since everything is an event now?

File: corebehrt/configs/outcomes_test.yaml [hunk 2]
```diff
@@ -8,15 +8,28 @@ paths:
...
-    match: [['M112']]
+    type: [ code ]
+    match: [ [ 'M112' ] ]
     match_how: startswith
     case_sensitive: false
-
+  TEST_COMBINED:
+    combinations:
+      primary:
+        type: [ code ]
```
Comment (Montgomeryyyy) [inline corebehrt/configs/outcomes_test.yaml:25]: when using the combined outcomes only the concepts_plus data is used and not the patient_info. So I think either type: code should not be an option as it is the only one, or it should also support mixing with patient info, like medication + death, etc

File: corebehrt/modules/cohort_handling/outcomes.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Montgomeryyyy) [reply corebehrt/modules/cohort_handling/outcomes.py:67]: haha I like that code rabbit can do reactions. But yeah could make sense to do.

File: corebehrt/modules/cohort_handling/outcomes.py [hunk 3]
```diff
@@ -64,18 +105,93 @@ def match_concepts(
...
         We then ensure all the types are true for a row by using bitwise_and.reduce. E.g. CONCEPT==COVID_TEST AND VALUE==POSITIVE
         """
+        # Handle empty DataFrame
+        if len(concepts_plus) == 0:
+            return create_empty_results_df()
+
+        # Make a copy to avoid modifying the original
+        filtered_concepts = concepts_plus.copy()
+
         if "exclude" in attrs:
```
Comment (Montgomeryyyy) [inline corebehrt/modules/cohort_handling/outcomes.py:115]: what is the use-case for exclusion?

**Review by Montgomeryyyy — CHANGES_REQUESTED** [review]
very cool, just minor things

File: corebehrt/configs/outcomes_test.yaml (hunk 2 above)
Comment (kirilklein) [reply corebehrt/configs/outcomes_test.yaml:25]: The idea behind type was to be able to select numeric value as well. But that is not implemented currently. I didn't want to rewrite the whole thing for now. 
We can do something more general like I did [here](https://github.com/kirilklein/PHAIR_EHR/blob/8c338ed55cd401bc5ac57c41cd63fd7aed294bbd/corebehrt/modules/cohort_handling/advanced/extract.py) or just switch to ACES to get the predicates csv and then convert to the format we have.

Comment (kirilklein) [reply corebehrt/configs/outcomes_test.yaml:25]: For now I'll just drop patients_info. We only ever used it for death, but that is an event now as well.

File: corebehrt/modules/cohort_handling/outcomes.py (hunk 3 above)
Comment (kirilklein) [reply corebehrt/modules/cohort_handling/outcomes.py:115]: E.g. all cancers except breas cancer
